lucene/core/src/test/org/apache/lucene/index/TestVectorValues.java

```diff
@@ -0,0 +1,650 @@
+        Document doc = new Document();
+        doc.add(new VectorField("dim", new float[4], ScoreFunction.DOT_PRODUCT));
+        w.addDocument(doc);
+      }
+      try (IndexWriter w2 = new IndexWriter(dir2, createIndexWriterConfig())) {
+        Document doc = new Document();
+        doc.add(new VectorField("dim", new float[5], ScoreFunction.DOT_PRODUCT));
+        w2.addDocument(doc);
+        try (DirectoryReader r = DirectoryReader.open(dir)) {
+          IllegalArgumentException expected = expectThrows(IllegalArgumentException.class,
+              () -> w2.addIndexes(new CodecReader[]{(CodecReader) getOnlyLeafReader(r)}));
+          assertEquals("cannot change vector dimension from 5 to 4 for field=\"dim\"", expected.getMessage());
+        }
+      }
+    }
+  }
+
+  public void testIllegalScoreFunctionChangeViaAddIndexesCodecReader() throws Exception {
+    try (Directory dir = newDirectory();
+         Directory dir2 = newDirectory()) {
+      try (IndexWriter w = new IndexWriter(dir, createIndexWriterConfig())) {
+        Document doc = new Document();
+        doc.add(new VectorField("dim", new float[4], ScoreFunction.DOT_PRODUCT));
+        w.addDocument(doc);
+      }
+      try (IndexWriter w2 = new IndexWriter(dir2, createIndexWriterConfig())) {
+        Document doc = new Document();
+        doc.add(new VectorField("dim", new float[4], ScoreFunction.EUCLIDEAN));
+        w2.addDocument(doc);
+        try (DirectoryReader r = DirectoryReader.open(dir)) {
+          IllegalArgumentException expected = expectThrows(IllegalArgumentException.class,
+              () -> w2.addIndexes(new CodecReader[]{(CodecReader) getOnlyLeafReader(r)}));
+          assertEquals("cannot change vector score function from EUCLIDEAN to DOT_PRODUCT for field=\"dim\"", expected.getMessage());
+        }
+      }
+    }
+  }
+
+  public void testIllegalDimChangeViaAddIndexesSlowCodecReader() throws Exception {
+    try (Directory dir = newDirectory();
+         Directory dir2 = newDirectory()) {
+      try (IndexWriter w = new IndexWriter(dir, createIndexWriterConfig())) {
+        Document doc = new Document();
+        doc.add(new VectorField("dim", new float[4], ScoreFunction.DOT_PRODUCT));
+        w.addDocument(doc);
+      }
+      try (IndexWriter w2 = new IndexWriter(dir2, createIndexWriterConfig())) {
+        Document doc = new Document();
+        doc.add(new VectorField("dim", new float[5], ScoreFunction.DOT_PRODUCT));
+        w2.addDocument(doc);
+        try (DirectoryReader r = DirectoryReader.open(dir)) {
+          IllegalArgumentException expected = expectThrows(IllegalArgumentException.class,
+              () -> TestUtil.addIndexesSlowly(w2, r));
+          assertEquals("cannot change vector dimension from 5 to 4 for field=\"dim\"", expected.getMessage());
+        }
+      }
+    }
+  }
+
+  public void testIllegalScoreFunctionChangeViaAddIndexesSlowCodecReader() throws Exception {
```

Comment:
We have good coverage of illegal `addIndexes` usage, but do we have a test case of a successful usage, adding in an index with vectors to an index without, and vice/versa, and then both?

Reply:
Ah, good idea, I will add